pkg/backup/backup.go

```diff
@@ -154,6 +165,13 @@ type backupContext struct {
 	// resource, from either the networking.k8s.io or extensions api groups. We only want to back them
 	// up once, from whichever api group we see first.
 	networkPoliciesBackedUp bool
+
+	// backedUpItems keeps track of items that have been backed up already. The string should be derived
+	// by calling String() on an itemKey instance representing the item.
+	backedUpItems sets.String
```

Comment:
I would probably do this as `map[itemKey]struct{}` - no need to stringify. And/or we could also create `type itemKeySet map[itemKey]struct{}` and add methods to it to make it more set like.

Reply:
heh, i had it that way originally (map[itemKey]struct{}) and switched to this to get the sets methods (mainly for clarity). Not having generics makes me sad at times like these.